### PR TITLE
Fixing query string in search results

### LINKED ISSUE #5156: Issue in search results of two or more words
We search with the input as two or more words, the first result seems to be fine but when I click the same or another button again such `All content` or `Notes` etc, the search bar and the results seems to include a plus sign `+` between the input words.

**Current behaviour:**
plissue2

**Expected behaviour:**

- [ ] The search bar and the `Results for` section should not include the `+` sign between the words on button clicks.

### PR DESCRIPTION
**Pull request by IshaGupta18**
Fixes #5156 

Closing #5236 and moving to this one. 

Using url_encode to achieve the required string for displaying search results.

image

### CODE REVIEW

**Comment by IshaGupta18** [comment]
@jywarren  @gauravano kindly review this PR if it's good to go. If yes, I would create follow up first-timer issues for the same. Thanks a lot!

**Comment by IshaGupta18** [comment]
Okay, so I see that the latest test written in `search_test.rb` is failing. @jywarren could you shed some light on it, please?

**Comment by gauravano** [comment]
Hey @IshaGupta18, this would just solve the issue for single tab. Instead of replacing it here, you can replace at https://github.com/publiclab/plots2/blob/master/app/views/search/_nav.html.erb#L3 as suggested here https://github.com/publiclab/plots2/pull/5236#pullrequestreview-218072935. Thanks!

File: app/views/search/all_content.html.erb [hunk 1]
```diff
@@ -3,12 +3,12 @@
 <div class="container">
   <% if  @nodes.all?{|k,v| v.blank?} %>
     <h4>
-      No Content found for <%= params[:query] %> :-(
+      No Content found for <%= CGI.unescape(params[:query]) %> :-(
```
Comment (IshaGupta18) [inline app/views/search/all_content.html.erb]: @gauravano this is what has been done to fix the string. In every tab. `params[:query]` is displayed. To make it universal for all tabs, we can make a function in the controller and return `CGI.unescape(params[:query])` from it, kind of like what has been done in the review, What do you think?

Comment (gauravano) [reply app/views/search/all_content.html.erb]: this line https://github.com/publiclab/plots2/blob/master/app/views/search/_nav.html.erb#L3 is the one which is calling the helper method generating all the nav links so just changing here from CGI.unescape to url_encode will do all the work. Try it out. T

Comment (IshaGupta18) [reply app/views/search/all_content.html.erb]: Does this work now?

Comment (gauravano) [reply app/views/search/all_content.html.erb]: Yes!

Comment (IshaGupta18) [reply app/views/search/all_content.html.erb]: Is it good to be merged?

**Comment by jywarren** [comment]
Thank you so much!!!! Sorry i wasn't able to help earlier @IshaGupta18 but you did a great job. Thanks to @gauravano as well!!!

**Pull request merged**